### ISSUE TITLE
Different answers on GPU and CPU in caption model

### ISSUE THREAD

**Issue by jiaoyunhao** [fchollet/keras#2902]
#a caption example model described in keras document
```
model = Sequential()
model.add(Merge([image_model, language_model], mode='concat', concat_axis=-1))
model.add(GRU(256, return_sequences=False))
model.add(Dense(vocab_size))
model.add(Activation('softmax'))
keras.optimizers.RMSprop(lr=0.5)
model.compile(loss='categorical_crossentropy', optimizer='rmsprop', metrics=['accuracy'])

images = np.load('images_n.npy')
images = images.astype("float32")
model.fit([images, partial_captions], next_word, batch_size = 20, nb_epoch=200)
```

When I run this on CPU, I got loss decreasing slightly. However, after moving to GPU, the loss kept increasing and I don't know why. Does anyone know how to figure it out?

**Comment by jiaoyunhao** [comment]
I have found that it's the version of keras that makes the difference. 1.0.2 and 1.0.3
